snmp/datadog_checks/snmp/discovery.py

```diff
@@ -0,0 +1,70 @@
+# (C) Datadog, Inc. 2020-present
+# All rights reserved
+# Licensed under Simplified BSD License (see LICENSE)
+
+import copy
+import json
+import time
+from typing import Any
+
+from datadog_checks.base import ConfigurationError
+
+from .compat import write_persistent_cache
+from .config import InstanceConfig
+
+
+def discover_instances(config, interval, check_ref):
+    # type: (InstanceConfig, float, Any) -> None
```

Comment:
I believe this should work:

```python
import weakref
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from .snmp import SnmpCheck

def discover_instances(config, interval, check_ref):
    # type: (InstanceConfig, float, weakref.ref[SnmpCheck]) -> None
```

This way we'll get type checking on the `check` object in this function.